include/deal.II/dofs/dof_handler_policy.h

```diff
@@ -69,7 +71,8 @@ namespace internal
          */
         virtual
         NumberCache
-        distribute_dofs (dealii::DoFHandler<dim,spacedim> &dof_handler) const = 0;
+        distribute_dofs (dealii::DoFHandler<dim,spacedim> &dof_handler,
+                         NumberCache &number_cache_current) const = 0;
```

Comment:
that will not work in the current hierarchy of classes. Keep in mind that marking of cells and calculation of local DoFs are done within DoFPolicy using functions in DoFTools, which in turn call methods of DoFHandler object which return data stored in that cache. So at the end of the day, we need to modify the current cache within the DoFPolicy to let external functions play well.